### ISSUE TITLE
Parse error with DataKinds

### ISSUE THREAD

**Issue by crockeea** [ndmitchell/hlint#100]
This simple code compiles with GHC:

```haskell
{-# LANGUAGE DataKinds #-}
module Foo where

type family ToList d = '[d]
```

but `hlint Foo.hs` complains:

```
testsuite/Foo.hs:4:20: Warning: Parse error: d
Found:
    module Foo where
    
  > type  ToList d = '[d]
```

I have hlint 1.9.12 and hse-1.16.0.1.
As a side note, why should a "parse error" ever have severity level "warning"? If a parse error doesn't constitute an error, I'm not sure what would.

**Comment by ndmitchell** [comment]
Should be fixed in the latest HSE/HLint combo, please reopen if not.
